### PR TITLE
MRG, FIX: Fix backref styling

### PR DESCRIPTION
**Pull request by larsoner**
So far just takes care of the first part of #587.

I plan to tackle the other problems, too, but this is a separable first step so this is ready for review/merge from my end @lucyleeow 

The change to `test_gen_gallery` was necessary because my computer username (`larsoner`) has `one` in it, so the regex was errantly matching paths it wasn't meant to. Using `basename` to get the filenames themselves fixes it.

### CODE REVIEW

**Comment by larsoner** [comment]
cc @drammock once this is in (or locally if you use my branch) you can hopefully try changing the MNE styling how you want

**Comment by larsoner** [comment]
Failure is just the nightly that we are currently ignoring

File: sphinx_gallery/tests/test_full.py [hunk 1]
```diff
@@ -183,23 +183,26 @@ def test_embed_links_and_styles(sphinx_app):
...
     assert 'numpy.ndarray.html' in lines
-    assert 'sphinx_gallery.backreferences.html#sphinx_gallery.backreferences.identify_names' in lines
+    assert 'sphinx_gallery.backreferences.html#sphinx_gallery.backreferences.identify_names' in lines  # noqa: E501
     # instances have an extra CSS class
-    assert 'class="sphx-glr-backref-module-matplotlib-figure sphx-glr-backref-type-py-class sphx-glr-backref-instance"><span class="n">x</span></a>' in lines  # noqa
-    assert 'class="sphx-glr-backref-module-matplotlib-figure sphx-glr-backref-type-py-class"><span class="n">Figure</span></a>' in lines  # noqa
+    assert 'class="sphx-glr-backref-module-matplotlib-figure sphx-glr-backref-type-py-class sphx-glr-backref-instance"><span class="n">x</span></a>' in lines  # noqa: E501
+    assert 'class="sphx-glr-backref-module-matplotlib-figure sphx-glr-backref-type-py-class"><span class="n">Figure</span></a>' in lines  # noqa: E501
+    # gh-587: no classes that are only marked as module without type
+    assert re.search(r'"sphx-glr-backref-module-\S*"', lines) is None
```
Comment (lucyleeow) [inline sphinx_gallery/tests/test_full.py]: Clever!

**Comment by lucyleeow** [comment]
Good job! Thanks @larsoner

**Pull request merged**